platform/app/src/routes/Mode/Mode.tsx

```diff
@@ -77,31 +106,47 @@ function defaultRouteInit(
     });
   });
 
-  // The hanging protocol matching service is fairly expensive to run multiple
-  // times, and doesn't allow partial matches to be made (it will simply fail
-  // to display anything if a required match fails), so we wait here until all metadata
-  // is retrieved (which will synchronously trigger the display set creation)
-  // until we run the hanging protocol matching service.
-
-  Promise.allSettled(allRetrieves).then(() => {
+  Promise.allSettled(allRetrieves).then(promises => {
     log.timeEnd(TimingEnum.STUDY_TO_DISPLAY_SETS);
     log.time(TimingEnum.DISPLAY_SETS_TO_FIRST_IMAGE);
     log.time(TimingEnum.DISPLAY_SETS_TO_ALL_IMAGES);
-    const displaySets = displaySetService.getActiveDisplaySets();
 
-    if (!displaySets || !displaySets.length) {
-      return;
+    const allPromises = [];
+    const remainingPromises = [];
+
+    function startRemainingPromises(remainingPromises) {
+      remainingPromises.forEach(p => p.forEach(p => p.start()));
     }
 
-    // Gets the studies list to use
-    const studies = getStudies(studyInstanceUIDs, displaySets);
+    function runHP() {
+      const { unsubscribe } = hangingProtocolService.subscribe(
+        hangingProtocolService.EVENTS.PROTOCOL_CHANGED,
+        () => {
+          startRemainingPromises(remainingPromises);
+          unsubscribe();
+        }
+      );
+      applyHangingProtocol();
+    }
```

Comment:
so there will be no remaining requests if the protocol does not change?

Reply:
I think Rodrigo meant to wait for the hanging protocol to be applied to start the remaining promises.

Reply:
Can you check if this gets called as soon as the hanging protocol initially applies?  That is, I think the HP changes when it first applies once after the initial set is done, so I think we are good.  Just would like to check that.

Reply:
Updated, thanks.